server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -108,6 +111,19 @@ public Coordinator(Settings settings, TransportService transportService, Allocat
         this.peerFinder = new CoordinatorPeerFinder(settings, transportService,
             new HandshakingTransportAddressConnector(settings, transportService), configuredHostsResolver);
         this.publicationHandler = new PublicationTransportHandler(transportService, this::handlePublishRequest, this::handleApplyCommit);
+        this.leaderChecker = new LeaderChecker(settings, transportService, new Runnable() {
```

Comment:
maybe put the runnable into a private helper method on Coordinator, so that we keep the constructor smaller

Reply:
I was debating whether to do so; this clinches it. I pushed 509e242a9da96ffc57bf96e9fcf95edd021cc0d8.